dashboard/client/src/pages/node/GRAMColumn.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box, Tooltip, Typography } from "@material-ui/core";
+import React from "react";
+import { RightPaddedTypography } from "../../common/CustomTypography";
+import PercentageBar from "../../components/PercentageBar";
+import { NodeDetail } from "../../type/node";
+import { Worker } from "../../type/worker";
+
+const GRAM_COL_WIDTH = 120;
+
+export const NodeGRAM = ({ node }: { node: NodeDetail }) => {
+  const nodeGRAMEntries = (node.gpus ?? []).map((gpu, i) => {
+    const props = {
+      gpuName: gpu.name,
+      utilization: gpu.memoryUsed,
+      total: gpu.memoryTotal,
+      slot: i,
+    };
+    return <GRAMEntry {...props} />;
+  });
+  return (
+    <div style={{ minWidth: 60 }}>
+      {nodeGRAMEntries.length === 0 ? (
+        <Typography color="textSecondary" component="span" variant="inherit">
+          N/A
+        </Typography>
+      ) : (
+        <div style={{ minWidth: GRAM_COL_WIDTH }}>{nodeGRAMEntries}</div>
+      )}
+    </div>
+  );
+};
+
+export const WorkerGRAM = ({
+  worker,
+  node,
+}: {
+  worker: Worker;
+  node: NodeDetail;
+}) => {
+  const workerGRAMEntries = (node.gpus ?? [])
+    .map((gpu, i) => {
```

Comment:
use reduce() to handle workerGRAMEntries instead of map().filter()

Reply:
What's the reason for the suggestion? I find map + filter to be easier to understand (better for code maintainability) than reduce functions.